transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/AttachmentHandlerImpl.kt

```diff
@@ -85,6 +85,11 @@ internal class AttachmentHandlerImpl(
     }
 
     override fun detach(token: String, attachmentId: String): DeleteAttachmentRequest? {
```

Comment:
- Add @Throws annotation to detach() function. 
- Also update the @Throws annotation of the MessagingClient.detach() as right now it's signature say that it throws ONLY IllegalStateException
- The comment is redundant.
- Create and use ErrorMessage.detachFailed() as the message parameter when you throw IllegalArgumentException instead of String. Reuse it later on in your unit test.